### PR TITLE
allow DocumentMetadata to hold arbirtary data

### PR DESCRIPTION
**Pull request by tanmaykm**
This introduces a new `custom` field in `DocumentMetadata` that is set to `nothing` by default, but can be used by user code to store arbirtary metadata against the document for use later. Having a pre-determined place to store such data would simplify processing in may cases.

### CODE REVIEW

**Comment by zgornel** [comment]
A more flexible approach would be to allow documents to hold any metadata (arbitrarily complex) and provide the mechanism for converting custom metadata to the 'standardized' `DocumentMetadata`  https://zgornel.github.io/StringAnalysis.jl/dev/doc_extensions/

Just for the record, there was a pull request extending `DocumentMetadata` with a few fields a while ago that went stale for months on end.

**Comment by tanmaykm** [comment]
Yes, having an abstract metadata type seems like a better idea. The API changes may be more intrusive though? Stemming depends on the language stored in metadata, that needs to be abstracted out. And there are a bunch of APIs in `metadata.jl`. Is there anything else?

**Comment by tanmaykm** [comment]
Rebased to resolve conflicts.

Probably this change will be sufficient for now? While we can continue discussing about a more appropriate metadata representation for the future.

File: test/document.jl [hunk 1]
```diff
@@ -1,6 +1,13 @@
 
 @testset "Document" begin
 
+    dmeta = TextAnalysis.DocumentMetadata(Languages.English(), "test title", "test author", "test time", Dict(:k1=>"v1", :k2=>"v2"))
+    @test (dmeta.language == Languages.English()) && (dmeta.title == "test title") && (dmeta.author == "test author") && (dmeta.timestamp == "test time") && (get(dmeta.custom, :k1, "") == "v1") && (get(dmeta.custom, :k2, "") == "v2")
+
+    # mutability
+    dmeta.custom = nothing
+    @test dmeta.custom == nothing
```
Comment (rssdev10) [inline test/document.jl:9]: `@test isnothing(dmeta.custom) ` or `@test dmeta.custom === nothing`

**Pull request merged**